templates/named.conf.options.erb

```diff
@@ -1,5 +1,5 @@
 options {
-	directory "/var/cache/bind";
+	directory "<%= scope.lookupvar('::dns::server::params::data_dir') %>";
```

Comment:
Changing this do data_dir will involve either changing the dns::server::options to accept another param or creating a local variable within the module that is assigned the ::dns::server::params::data_dir variable.

Overriding this is already possible by using hiera and passing in params into the params module.

Your call ...